Terminal.Gui/Views/ScrollBarView.cs

```diff
@@ -38,74 +37,13 @@ public class ScrollBarView : View
     /// </summary>
     public ScrollBarView ()
```

Comment:
> I think:
> 
> * the v2 class should be named `ScrollBar`

I agree and I'll do it i this PR.

> * An instance of `ScrollBar` should act 100% independently of whether there is "another scrollbar" or not. Any logic that coordinates two scrollbars should be outside of the `ScrollBar` class.

To do this, simply start each view that wants to use separate scroll bars such as ScrollBarType = ScrollBarType.Vertical or ScrollBarType = ScrollBarType.Horizontal and manipulate what each one should do differently from the other.

> * Logic for "auto hiding" scroll bars should NOT be in `ScrollBar`. External code should be able to just do `scrollBar.Visible =`.

The `Visible` property is actually the one that manipulates its visibility and its value is constantly changed whenever its values are resized or reset. Therefore, in my opinion it should only be used to manipulate the visibility imposed by auto-hide. But the user may not want to draw the scroll bar even if it is necessary or want to draw it even if it is not necessary. To enforce this condition, there are the `AutoHideScrollBars` and `ShowScrollIndicator` properties. If `AutoHideScrollBars` is `false` you can manipulate the visibility at will using the `ShowScrollIndicator` property. That's why the `Visible` property is reserved for the scroll bar so that it can be manipulated at will when the `AutoHideScrollBars` property is `true`. Therefore, it is not worth manipulating the `Visible` property directly.

> * The public API `ScrollBar` exposes should be primarily "set these properties and subscribe to these events"; Normal use should not require overriding methods.

Yes, the API should explain how to use the scroll bar in a view. Of course, `TextView` and `TableView` are more complex examples because they already have automatic management of these adjustments defined, as they can be used without a scroll bar.
With this PR it is now possible to use negative limits so that a view does not need to implement left and top offsets, but only provides the size of the columns and rows. See the `ScrollBarBuiltIn` scenario.

Reply:
> the v2 class should be named ScrollBar

Done.